gk/main.c

```diff
@@ -573,6 +639,12 @@ gk_proc(void *arg)
 
 	gk_conf_hold(gk_conf);
 
+	if (block_idx == 0) {
+		ret = gk_hold_arp_and_nd(gk_conf, lcore);
```

Comment:
Notice that although unlikely, a Grantor server could be in the back network of a Gatekeeper server, that is, the packet would be sent directly to the Grantor server; no gateway would be needed in this case.